Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionMediaSourceAVFObjC.h

```diff
@@ -68,8 +68,9 @@ ALLOW_NEW_API_WITHOUT_GUARDS_BEGIN
     void audioRendererDidReceiveError(AVSampleBufferAudioRenderer *, NSError *, bool& shouldIgnore) override;
 ALLOW_NEW_API_WITHOUT_GUARDS_END
 
-    void addSourceBuffer(SourceBufferPrivateAVFObjC*);
-    void removeSourceBuffer(SourceBufferPrivateAVFObjC*);
+    enum class ShouldAddParser : bool { No, Yes };
+    void addSourceBuffer(SourceBufferPrivateAVFObjC*, ShouldAddParser);
```

Comment:
this code path (existing) isn't particularly great. We have the SourceBuffer calling into the CDMSession only to have the CDMSession make a re-entrant call into the SourceBuffer to get the AVStreamDataParser. The SourceBuffer knows the information to start.